cora/scripts/makesky.py

```diff
@@ -13,93 +13,241 @@
-        raise click.ClickException("Output file %s already exists" % filename)
+    def __init__(self):
+
+        # Set the CHIME band as the internal default
+        self.freq = (800.0, 400.0, 1025)
+
+        self.channel_range = None
+        self.channel_list = None
+        self.channel_bin = 1
+        self.freq_mode = "centre"
+
+    @property
+    def frequencies(self):
+        """The frequency centres in MHz."""
+        return self._calculate()[0]
+
+    @property
+    def freq_width(self):
+        """The frequency width in MHz."""
+        return self._calculate()[1]
+
+    def _calculate(self):
+        """Calculate the frequencies from the parameters."""
+        # Generate the set of frequency channels given the parameters
+
+        sf, ef, nf = self.freq
+        if self.freq_mode == "centre":
+            df = abs((ef - sf) / (nf - 1))
+            frequencies = np.linspace(sf, ef, nf, endpoint=True)
+        else:
+            df = (ef - sf) / nf
+            frequencies = sf + df * (np.arange(nf) + 0.5)
+
+        # Select a subset of channels if required
+        if self.channel_list is not None:
+            frequencies = frequencies[self.channel_list]
+        elif self.channel_range is not None:
+            frequencies = frequencies[self.channel_range[0] : self.channel_range[1]]
+
+        # Rebin frequencies if needed
+        if self.channel_bin > 1:
+            frequencies = frequencies.reshape(-1, self.channel_bin).mean(axis=1)
+            df = df * self.channel_bin
+
+        return frequencies, df
+
+    @classmethod
+    def _set_attr(cls, ctx, param, value):
+        state = ctx.ensure_object(cls)
+        setattr(state, param.name, value)
+        return value
+
+    @classmethod
+    def options(cls, f):
+
+        FREQ = ListOfType("frequency list", int)
+
+        options = [
+            click.option(
+                "--freq",
+                help=(
+                    "Define the frequency channelisation, give the start and stop "
+                    "frequencies (in MHz) and the effective number of channels. "
+                    "Default is for CHIME: FSTART=800.0, FSTOP=400.0, FNUM=1025"
+                ),
+                metavar="FSTART FSTOP FNUM",
+                type=(float, float, int),
+                default=(800.0, 400.0, 1025),
+                expose_value=False,
+                callback=cls._set_attr,
+            ),
+            click.option(
+                "--channel-range",
+                help="Select a range of frequency channels. Overriden by channel range.",
+                type=(int, int),
+                metavar="CSTART CSTOP",
+                default=(None, None),
+                expose_value=False,
+                callback=cls._set_attr,
+            ),
+            click.option(
+                "--channel-list",
+                help="Select a list of frequency channels. Takes priority over channel range.",
+                type=FREQ,
```

Comment:
I think this should be type(list) because you are giving it a list of frequency indices (see line 88). metavar could be CHANNEL_LIST

Reply:
I think this is fine as `FREQ` is defined to be a list of indices of L108.